filebeat/input/event.go

```diff
@@ -39,8 +39,19 @@ func (e *Event) ToMapStr() common.MapStr {
 		"input_type":            e.InputType,
 	}
 
-	if e.JSONConfig != nil && len(e.JSONFields) > 0 {
-		mergeJSONFields(e, event)
+	// Add data fields which are added by the readers
+	for key, value := range e.Data {
+		event[key] = value
+	}
+
+	// Check if json fields exist
+	var jsonFields common.MapStr
+	if fields, ok := event["json"]; ok {
+		jsonFields = fields.(common.MapStr)
+	}
```

Comment:
why not have `Data` and `JsonFields` (or `JSON`) in message type? This handling of json by unpacking from event + delete key seems a little contrived.